packages/flutter/lib/src/gestures/binding.dart

```diff
@@ -99,8 +182,41 @@ mixin GestureBinding on BindingBase implements HitTestable, HitTestDispatcher, H
 
   void _flushPointerEventQueue() {
     assert(!locked);
-    while (_pendingPointerEvents.isNotEmpty)
-      _handlePointerEvent(_pendingPointerEvents.removeFirst());
+
+    // Stop resampler if resampling is not enabled. This is a no-op if
+    // resampling was never enabled.
+    if (!resamplingEnabled) {
```

Comment:
This looks like better in the setter of `resamplingEnabled`.

Reply:
I think this is put here for the `assert(!locked)` condition as `stop` will fire the remaining events.

Reply:
we need this here to be sure that assert(!locked)